src/index/ivf/ivf_config.h

```diff
@@ -86,6 +92,25 @@ class ScannConfig : public IvfFlatConfig {
 
         return Status::success;
     }
+
+    inline Status
+    CheckAndAdjustForRangeSearch(std::string* err_msg) override {
+        if (!faiss::support_pq_fast_scan) {
+            *err_msg = "scann index can not be used on current cpu, need at least avx2 instruction set";
+            LOG_KNOWHERE_ERROR_ << *err_msg;
+            return Status::invalid_instruction_set;
+        }
+        return Status::success;
+    }
+
+    inline Status
+    CheckAndAdjustForBuild() override {
+        if (!faiss::support_pq_fast_scan) {
+            LOG_KNOWHERE_ERROR_ << "scann index can not be used on current cpu, need at least avx2 instruction set";
```

Comment:
"the current index is not supported on the current CPU model. AVX2 support is needed for x86."